src/depfile.rs

```diff
@@ -36,10 +36,17 @@ fn read_path<'a>(scanner: &mut Scanner<'a>) -> ParseResult<Option<&'a str>> {
     let start = scanner.ofs;
     loop {
         match scanner.read() {
-            '\0' | ' ' | ':' | '\r' | '\n' => {
+            '\0' | ' ' | '\r' | '\n' => {
                 scanner.back();
                 break;
             }
+            ':' => {
+                let peek = scanner.peek();
+                if peek != '\\' && peek != '/' {
```

Comment:
Could you do this check only at the very start of the path? To avoid including colons in the middle of strings, as I think technically `out.o:/some/absolute/path.c` is a valid depfile.

Reply:
Looking at the GNU Make docs, it appears it attempts to detect drive letters specifically, and I verified it does allow the spaceless syntax you mention.

On the other hand, depfiles are only sorta Makefiles.  I vaguely recall there are a bunch of weird rules particularly around end of line handling where the rules are a bunch of nonsense.  I imagine (hope?) any tool generating these uses spaces, in particular because it's otherwise impossible to generate a spaceless depfile for a single-character output file.

So I hope it's harmless to do the simple thing I merged in 738e5641f169d96ac634c9437373a9ce4f1f6c61